src/test/java/ru/hogwarts/school/controller/controller_mockmvc/FacultyControllerTest.java

```diff
@@ -0,0 +1,108 @@
+    @Autowired
+    ObjectMapper objectMapper;
+    @Test
+    void getById() throws Exception {
+        Faculty faculty = new Faculty(1L,"dgdfg","red");
+        when(facultyRepository.findById(1L)).thenReturn(Optional.of(faculty));
+        mockMvc.perform(MockMvcRequestBuilders.get("/faculty/1")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.name").value("dgdfg"))
+                .andExpect(jsonPath("$.color").value("red"));
+    }
+    @Test
+    void create() throws Exception {
+        Faculty faculty = new Faculty(1L,"dgdfg","red");
+        when(facultyRepository.save(any(Faculty.class))).thenReturn(faculty);
+        mockMvc.perform(post("/faculty")
+                        .content(objectMapper.writeValueAsString(faculty))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.name").value("dgdfg"))
+                .andExpect(jsonPath("$.color").value("red"));
+    }
+    @Test
+    void update() throws Exception {
+        Faculty faculty = new Faculty(1L,"dgdfg","red");
+        when(facultyRepository.findById(1L)).thenReturn(Optional.of(faculty));
+        when(facultyRepository.save(any(Faculty.class))).thenReturn(faculty);
+        mockMvc.perform(put("/faculty/"+faculty.getId())
+                        .content(objectMapper.writeValueAsString(faculty))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.name").value("dgdfg"))
+                .andExpect(jsonPath("$.color").value("red"));
+    }
+    @Test
+    void delete() throws Exception {
+        Faculty faculty = new Faculty(1L,"dgdfg","red");
+        when(facultyRepository.findById(1L)).thenReturn(Optional.of(faculty));
+
+        mockMvc.perform(MockMvcRequestBuilders.delete("/faculty/" + faculty.getId())
+                        .content(objectMapper.writeValueAsString(faculty))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.name").value("dgdfg"))
+                .andExpect(jsonPath("$.color").value("red"));
+    }
+    @Test
+    void byColorOrNameTest() throws Exception {
+        when(facultyRepository.findAllByColorLikeIgnoreCaseOrNameLikeIgnoreCase("red", "den")).thenReturn(Arrays.asList(
```

Comment:
```suggestion
        when(facultyRepository.findAllByColorLikeIgnoreCaseOrNameLikeIgnoreCase("den", "den")).thenReturn(Arrays.asList(
```
Ты же ищешь по "den", поэтому значение нужно указать такое же.